File: Unit test/DivisionWithExceptions/C++/Delenie.cpp
#include <iostream>

double divide(int, int);

int main()
{
	std::cout << "Enter the first number";
	double num1;
	std::cin >> num1;

	std::cout << "Enter the second number";
	double num2;
	std::cin >> num2;
	try
	{
	  double res = divide(num1, num2);
		std::cout << res << std::endl;
	}
	catch (const std::exception& err)
	{
		std::cout << "Error!!!" << std::endl;
	}
	std::cout << "The End..." << std::endl;
	return 0;
}

double divide(int num1, int num2)
{
	if (num2 == 0)
		throw std::exception();
	return num1 / num2;
}

File: Unit test/DivisionWithExceptions/Java/Delenie.java
package com.company;
import java.util.Scanner;
public class Main
{
    public static void main(String[] args)
    {
        try
        {
            Scanner in = new Scanner(System.in);

            System.out.println("Введите первое число");
            double num1 = in.nextDouble();

            System.out.println("Введите второе число");
            double num2 = in.nextDouble();

            if(num2 == 0)
            {
             System.out.println ("Вы не можете делить на 0");
            }
            else
            {

                double res = num1 / num2;
                System.out.print("Результат: ");
                System.out.println(res);
            }
        }
        catch (Exception err)
        {
            System.out.println("Ошибка ввода");
        }
        finally
        {
            System.out.println("Конец");
        }

    }


}
